### PR TITLE
CIRC-886 Truncate due date based on patron expiration

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Approach 
- Truncate the item’s due date and time to the latest date and time where the service point is open, and the user is not expired;
- Add tests;

Resolves: https://issues.folio.org/browse/CIRC-886

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: src/test/java/api/loans/CheckOutByBarcodeTests.java [hunk 1]
```diff
@@ -1336,6 +1337,26 @@ public void checkOutFailsWhenEventPublishingFailsWithBadRequestError() {
       "Error during publishing Event Message in PubSub. Status code: 400"));
   }
 
+  @Test
+  public void dueDateShouldBeTruncatedToTheEndOfLastWorkingDayBeforePatronExpiration() {
+    final UUID book = materialTypesFixture.book().getId();
+
+    circulationRulesFixture.updateCirculationRules(
+      createRulesWithFixedDueDateInLoanPolicy( "m " + book));
```
Comment (alexanderkurash) [inline src/test/java/api/loans/CheckOutByBarcodeTests.java:1345]: Space

File: src/test/java/api/loans/CheckOutByBarcodeTests.java [hunk 2]
```diff
@@ -1336,6 +1337,26 @@ public void checkOutFailsWhenEventPublishingFailsWithBadRequestError() {
...
+
+    IndividualResource firstBookTypeItem = itemsFixture.basedUponNod();
```
Comment (alexanderkurash) [inline src/test/java/api/loans/CheckOutByBarcodeTests.java:1347]: IndividualResource item ?

File: src/test/java/api/support/fixtures/CalendarExamples.java [hunk 3]
```diff
@@ -62,6 +64,9 @@
   public static final LocalDate CASE_FRI_SAT_MON_DAY_ALL_PREV_DATE = new LocalDate(2018, 12, 14);
   public static final LocalDate CASE_FRI_SAT_MON_DAY_ALL_CURRENT_DATE = new LocalDate(2018, 12, 15);
   public static final LocalDate CASE_FRI_SAT_MON_DAY_ALL_NEXT_DATE = new LocalDate(2018, 12, 17);
+  public static final LocalDate FIRST_DAY_IN_A_DAY = new LocalDate(LocalDate.now().plusDays(2));
+  public static final LocalDate SECOND_DAY_IN_A_DAY = new LocalDate(LocalDate.now().plusDays(3));
+  public static final LocalDate THIRD_DAY_IN_A_DAY = new LocalDate(LocalDate.now().plusDays(4));
```
Comment (alexanderkurash) [inline src/test/java/api/support/fixtures/CalendarExamples.java:69]: Can we use ClockManager?

File: src/test/java/api/support/fixtures/CalendarExamples.java [hunk 4]
```diff
@@ -34,6 +34,8 @@
   public static final String CASE_PREV_OPEN_AND_CURRENT_NEXT_CLOSED = "85346678-2f09-4bc9-8924-3734882d44a3";
   public static final String CASE_CALENDAR_IS_EMPTY_SERVICE_POINT_ID = "66655555-2f09-4bc9-8924-3734882d44a3";
 
+  public static final String CASE_IN_ONE_DAY_IS_OPEN_NEXT_TWO_DAYS_CLOSED = "6ab38b7a-c889-4839-a337-86aad0297d7c";
```
Comment (alexanderkurash) [inline src/test/java/api/support/fixtures/CalendarExamples.java:37]: I think we can remove _IN from this name

File: src/test/java/api/loans/CheckOutByBarcodeTests.java [hunk 5]
```diff
@@ -1336,6 +1337,26 @@ public void checkOutFailsWhenEventPublishingFailsWithBadRequestError() {
...
+    IndividualResource steve = usersFixture.steve(user -> user.expires(
+      DateTime.now().plusDays(3)));
+
+    JsonObject response = checkOutFixture.checkOutByBarcode(
+      new CheckOutByBarcodeRequestBuilder()
+        .forItem(firstBookTypeItem)
+        .to(steve)
+        .at(CASE_IN_ONE_DAY_IS_OPEN_NEXT_TWO_DAYS_CLOSED)).getJson();
+
+    assertThat(response.getString("dueDate"), is("2020-10-29T23:59:59.000Z"));
```
Comment (alexanderkurash) [inline src/test/java/api/loans/CheckOutByBarcodeTests.java:1357]: Constant date?

File: src/test/java/api/loans/CheckOutByBarcodeTests.java (hunk 1 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/CheckOutByBarcodeTests.java]: fixed

File: src/test/java/api/loans/CheckOutByBarcodeTests.java (hunk 2 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/CheckOutByBarcodeTests.java]: renamed

File: src/test/java/api/support/fixtures/CalendarExamples.java (hunk 3 above)
Comment (roman-barannyk) [reply src/test/java/api/support/fixtures/CalendarExamples.java]: updated

File: src/test/java/api/support/fixtures/CalendarExamples.java (hunk 4 above)
Comment (roman-barannyk) [reply src/test/java/api/support/fixtures/CalendarExamples.java]: renamed

File: src/test/java/api/loans/CheckOutByBarcodeTests.java (hunk 5 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/CheckOutByBarcodeTests.java]: fixed

**Review by OleksandrVidinieiev — APPROVED** [review]

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**